### ISSUE TITLE
Embedded struct Support

### ISSUE THREAD

**Issue by wayne-brush-optii** [go-swagger/go-swagger#2027]
I was wondering if it was planned to support embedded structures (inheritance). I noticed Issue #413 gives the specifics for this issue but has lived for over 3 years. Does this mean that this tool will not support this functionality.

**Comment by casualjim** [comment]
embedded structs should work
discriminator values it won't create but you can start with a spec and deep merge as nuclear option

**Comment by casualjim** [comment]
take a look here: https://github.com/go-swagger/go-swagger/blob/master/fixtures/goparsing/classification/models/nomodel.go#L478-L614

**Comment by wayne-brush-optii** [comment]
Thanks for the examples. The only problem is that this solves the problem for packages I create but any package I import (3rd party) would then have to support this capability for it to work.

**Comment by ivanovaleksey** [comment]
@casualjim @fredbi embedding 3rd-party types seems to work fine. 
However there is a case with alias on 3rd-party which seems to be broken.

Consider the following structure

```
# github.com/foo/a
color
  color.go
    type Color struct { Title string }

    type SamePackageAlias Color

alias.go
  type AnotherPackageAlias color.Color
```

```
# github.com/foo/b

import (
  "github.com/foo/a"
  "github.com/foo/a/color"
)

type Figure struct {
   a.AnotherPackageAlias // won't be rendered in Figure definition

   color.Color // would be rendered in Figure definition

   color.SamePackageAlias // would be rendered in Figure definition

   AnyName a.Color // also would be rendered in Figure definition
}
```

The problem is combination of embedding + aliasing in another package.
I can create demo repos if that would help.

In a real project I have alias with the same name. I created demo repo and figured out that it same naming is the cause. I opened issue #2416 for that.
Then I renamed alias in a real project but it didn't help. This time I figured out that there is this corner case of embedding + aliasing in another package.

Is it can be fixed?
Please tell me if I should open another issue.

**Comment by casualjim** [comment]
Yeah that's a missing feature, we never implemented type aliasing because it didn't exist when all of this was written originally.

So I think it needs a new issue, not just a comment

**Comment by ivanovaleksey** [comment]
@casualjim thanks for the reply. I would like to note that most aliasing cases seem to work fine. It's alias in another package that doesn't work.
I opened new issue https://github.com/go-swagger/go-swagger/issues/2417
